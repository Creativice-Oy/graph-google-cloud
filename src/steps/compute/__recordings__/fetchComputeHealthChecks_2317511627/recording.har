{
  "log": {
    "_recordingName": "fetchComputeHealthChecks",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 584,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 584,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 19 Feb 2021 14:54:05 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-02-19T14:54:05.306Z",
        "time": 181,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 181
        }
      },
      {
        "_id": "5a19208022049d78b9012afb922618c9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "compute.googleapis.com"
            }
          ],
          "headersSize": 564,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://compute.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v2/global/healthChecks"
        },
        "response": {
          "bodySize": 748,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 748,
            "text": "{\"id\":\"projects/j1-gc-integration-dev-v2/global/healthChecks\",\"items\":[{\"id\":\"6034280886294286826\",\"creationTimestamp\":\"2021-02-17T16:04:21.342-08:00\",\"name\":\"load-balancer-health-check\",\"checkIntervalSec\":1,\"timeoutSec\":1,\"unhealthyThreshold\":2,\"healthyThreshold\":2,\"type\":\"TCP\",\"tcpHealthCheck\":{\"port\":80,\"proxyHeader\":\"NONE\"},\"selfLink\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v2/global/healthChecks/load-balancer-health-check\",\"kind\":\"compute#healthCheck\"}],\"selfLink\":\"https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev-v2/global/healthChecks\",\"kind\":\"compute#healthCheckList\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "1E9ZeXetDFxDKtxa5vtB4FRtix0=/r2r6Kjuinbr3aTHy-vznU8YAdoo="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 19 Feb 2021 14:54:05 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 553,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-02-19T14:54:05.499Z",
        "time": 292,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 292
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
