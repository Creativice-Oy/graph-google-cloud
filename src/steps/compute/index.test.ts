jest.setTimeout(60000);

import {
  Recording,
  createMockStepExecutionContext,
} from '@jupiterone/integration-sdk-testing';
import { setupGoogleCloudRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import { fetchComputeDisks, fetchComputeInstances } from '.';
import { integrationConfig } from '../../../test/config';
import {
  ENTITY_TYPE_COMPUTE_DISK,
  ENTITY_TYPE_COMPUTE_INSTANCE,
} from './constants';

describe('#fetchComputeDisks', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchComputeDisks',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchComputeDisks(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(context.jobState.collectedEntities).toMatchGraphObjectSchema({
      _class: ['DataStore'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_compute_disk' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          description: { type: 'string' },
          zone: { type: 'string' },
          sizeGB: { type: 'string' },
          status: { type: 'string' },
          sourceImage: { type: 'string' },
          sourceImageId: { type: 'string' },
          type: { type: 'string' },
          licenses: {
            type: 'array',
            items: { type: 'string' },
          },
          guestOsFeatures: {
            type: 'array',
            items: { type: 'string' },
          },
          lastAttachTimestamp: { type: 'number' },
          labelFingerprint: { type: 'string' },
          licenseCodes: {
            type: 'array',
            items: { type: 'string' },
          },
          physicalBlockSizeBytes: { type: 'string' },
          kind: { type: 'string' },
          encrypted: true,
        },
      },
    });
  });
});

describe('#fetchComputeInstances', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchComputeInstances',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchComputeDisks(context);
    await fetchComputeInstances(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === ENTITY_TYPE_COMPUTE_DISK,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['DataStore'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_compute_disk' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          description: { type: 'string' },
          zone: { type: 'string' },
          sizeGB: { type: 'string' },
          status: { type: 'string' },
          sourceImage: { type: 'string' },
          sourceImageId: { type: 'string' },
          type: { type: 'string' },
          licenses: {
            type: 'array',
            items: { type: 'string' },
          },
          guestOsFeatures: {
            type: 'array',
            items: { type: 'string' },
          },
          lastAttachTimestamp: { type: 'number' },
          labelFingerprint: { type: 'string' },
          licenseCodes: {
            type: 'array',
            items: { type: 'string' },
          },
          physicalBlockSizeBytes: { type: 'string' },
          kind: { type: 'string' },
          encrypted: true,
        },
      },
    });

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === ENTITY_TYPE_COMPUTE_INSTANCE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Host'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_compute_instance' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          machineType: { type: 'string' },
          status: { type: 'string' },
          zone: { type: 'string' },
          canIpForward: { type: 'boolean' },
          cpuPlatform: { type: 'string' },
          labelFingerprint: { type: 'string' },
          startRestricted: { type: 'boolean' },
          deletionProtection: { type: 'boolean' },
          fingerprint: { type: 'string' },
          kind: { type: 'string' },
        },
      },
    });

    const computeInstanceUsesDiskRelationship = context.jobState.collectedRelationships.filter(
      (r) => r._type === 'google_compute_instance_uses_disk',
    );

    expect(computeInstanceUsesDiskRelationship).toEqual(
      computeInstanceUsesDiskRelationship.map((r) =>
        expect.objectContaining({
          _class: 'USES',
        }),
      ),
    );
  });
});
