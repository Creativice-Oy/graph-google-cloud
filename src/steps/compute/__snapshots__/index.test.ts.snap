// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchComputeDisks should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "DataStore",
        "Disk",
      ],
      "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "creationTimestamp": "2020-08-12T08:20:05.930-07:00",
            "guestOsFeatures": Array [
              Object {
                "type": "VIRTIO_SCSI_MULTIQUEUE",
              },
            ],
            "id": "6905138577447433802",
            "kind": "compute#disk",
            "labelFingerprint": "42WmSpB8rSM=",
            "lastAttachTimestamp": "2020-08-12T08:20:05.930-07:00",
            "licenseCodes": Array [
              "1000205",
            ],
            "licenses": Array [
              "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
            ],
            "name": "testvm",
            "physicalBlockSizeBytes": "4096",
            "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
            "sizeGb": "10",
            "sourceImage": "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-9-stretch-v20200805",
            "sourceImageId": "6709658075886210235",
            "status": "READY",
            "tags": undefined,
            "type": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/diskTypes/pd-standard",
            "users": Array [
              "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
            ],
            "zone": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a",
          },
        },
      ],
      "_type": "google_compute_disk",
      "active": true,
      "classification": null,
      "createdOn": 1597245605930,
      "description": undefined,
      "displayName": "testvm",
      "encrypted": true,
      "guestOsFeatures": Array [
        "VIRTIO_SCSI_MULTIQUEUE",
      ],
      "id": "6905138577447433802",
      "kind": "compute#disk",
      "labelFingerprint": "42WmSpB8rSM=",
      "lastAttachTimestamp": 1597245605930,
      "licenseCodes": Array [
        "1000205",
      ],
      "licenses": Array [
        "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
      ],
      "name": "testvm",
      "physicalBlockSizeBytes": "4096",
      "sizeGB": "10",
      "sourceImage": "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-9-stretch-v20200805",
      "sourceImageId": "6709658075886210235",
      "status": "READY",
      "type": "pd-standard",
      "zone": "us-central1-a",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_compute_disk",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;

exports[`#fetchComputeInstances should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "DataStore",
        "Disk",
      ],
      "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "creationTimestamp": "2020-08-12T08:20:05.930-07:00",
            "guestOsFeatures": Array [
              Object {
                "type": "VIRTIO_SCSI_MULTIQUEUE",
              },
            ],
            "id": "6905138577447433802",
            "kind": "compute#disk",
            "labelFingerprint": "42WmSpB8rSM=",
            "lastAttachTimestamp": "2020-08-12T08:20:05.930-07:00",
            "licenseCodes": Array [
              "1000205",
            ],
            "licenses": Array [
              "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
            ],
            "name": "testvm",
            "physicalBlockSizeBytes": "4096",
            "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
            "sizeGb": "10",
            "sourceImage": "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-9-stretch-v20200805",
            "sourceImageId": "6709658075886210235",
            "status": "READY",
            "tags": undefined,
            "type": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/diskTypes/pd-standard",
            "users": Array [
              "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
            ],
            "zone": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a",
          },
        },
      ],
      "_type": "google_compute_disk",
      "active": true,
      "classification": null,
      "createdOn": 1597245605930,
      "description": undefined,
      "displayName": "testvm",
      "encrypted": true,
      "guestOsFeatures": Array [
        "VIRTIO_SCSI_MULTIQUEUE",
      ],
      "id": "6905138577447433802",
      "kind": "compute#disk",
      "labelFingerprint": "42WmSpB8rSM=",
      "lastAttachTimestamp": 1597245605930,
      "licenseCodes": Array [
        "1000205",
      ],
      "licenses": Array [
        "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
      ],
      "name": "testvm",
      "physicalBlockSizeBytes": "4096",
      "sizeGB": "10",
      "sourceImage": "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-9-stretch-v20200805",
      "sourceImageId": "6709658075886210235",
      "status": "READY",
      "type": "pd-standard",
      "zone": "us-central1-a",
    },
    Object {
      "_class": Array [
        "Host",
      ],
      "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "canIpForward": false,
            "cpuPlatform": "Intel Haswell",
            "creationTimestamp": "2020-08-12T08:20:05.921-07:00",
            "deletionProtection": false,
            "disks": Array [
              Object {
                "autoDelete": true,
                "boot": true,
                "deviceName": "persistent-disk-0",
                "diskSizeGb": "10",
                "guestOsFeatures": Array [
                  Object {
                    "type": "VIRTIO_SCSI_MULTIQUEUE",
                  },
                ],
                "index": 0,
                "interface": "SCSI",
                "kind": "compute#attachedDisk",
                "licenses": Array [
                  "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
                ],
                "mode": "READ_WRITE",
                "source": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
                "type": "PERSISTENT",
              },
              Object {
                "autoDelete": true,
                "boot": false,
                "deviceName": "local-ssd-0",
                "diskSizeGb": "375",
                "index": 1,
                "interface": "SCSI",
                "kind": "compute#attachedDisk",
                "mode": "READ_WRITE",
                "type": "SCRATCH",
              },
            ],
            "fingerprint": "4gmEF7tlfn8=",
            "id": "2248191918776890954",
            "kind": "compute#instance",
            "labelFingerprint": "42WmSpB8rSM=",
            "machineType": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/machineTypes/n1-standard-1",
            "metadata": Object {
              "fingerprint": "9yLGkjCLBTs=",
              "items": Array [
                Object {
                  "key": "foo",
                  "value": "bar",
                },
                Object {
                  "key": "startup-script",
                  "value": "echo Hello",
                },
              ],
              "kind": "compute#metadata",
            },
            "name": "testvm",
            "networkInterfaces": Array [
              Object {
                "fingerprint": "RpvQNHzHRp8=",
                "kind": "compute#networkInterface",
                "name": "nic0",
                "network": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/networks/default",
                "networkIP": "10.128.0.2",
                "subnetwork": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/us-central1/subnetworks/default",
              },
            ],
            "scheduling": Object {
              "automaticRestart": true,
              "onHostMaintenance": "MIGRATE",
              "preemptible": false,
            },
            "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
            "serviceAccounts": Array [
              Object {
                "email": "j1-gc-integration-dev-sa-tf@j1-gc-integration-dev.iam.gserviceaccount.com",
                "scopes": Array [
                  "https://www.googleapis.com/auth/compute.readonly",
                  "https://www.googleapis.com/auth/devstorage.read_only",
                  "https://www.googleapis.com/auth/userinfo.email",
                ],
              },
            ],
            "startRestricted": false,
            "status": "RUNNING",
            "tags": Array [
              Object {
                "testvmtag1": true,
              },
              Object {
                "testvmtag2": true,
              },
            ],
            "zone": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a",
          },
        },
      ],
      "_type": "google_compute_instance",
      "active": true,
      "canIpForward": false,
      "cpuPlatform": "Intel Haswell",
      "createdOn": 1597245605921,
      "deletionProtection": false,
      "displayName": "testvm",
      "fingerprint": "4gmEF7tlfn8=",
      "id": "2248191918776890954",
      "kind": "compute#instance",
      "labelFingerprint": "42WmSpB8rSM=",
      "machineType": "n1-standard-1",
      "name": "testvm",
      "startRestricted": false,
      "status": "RUNNING",
      "zone": "us-central1-a",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "USES",
      "_fromEntityKey": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
      "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm|uses|https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
      "_toEntityKey": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
      "_type": "google_compute_instance_uses_disk",
      "autoDelete": true,
      "deviceName": "persistent-disk-0",
      "displayName": "USES",
      "interface": "SCSI",
      "mode": "READ_WRITE",
    },
  ],
  "encounteredTypes": Array [
    "google_compute_disk",
    "google_compute_instance",
    "google_compute_instance_uses_disk",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 1,
}
`;
